### PR TITLE
Adding `sma_will_rise(days=15)` as new label

### PR DESCRIPTION
**Pull request by tedsmoore**
Added a new label condition: SMA will rise after a given number of days (default=15). This seems like a reasonable, somewhat useful thing we can possibly predict based on trends alone.

### CODE REVIEW

File: src/features/build_features.py [hunk 1]
```diff
@@ -1,65 +1,85 @@
...
+    X, y = b.run()
+    """
+
     def __init__(self, feed, instrument):
         strategy.BacktestingStrategy.__init__(self, feed)
-        self.__instrument = instrument
-        self.__sma = ma.SMA(feed[instrument].getAdjCloseDataSeries(), 15)
-        self.__ema = ma.EMA(feed[instrument].getAdjCloseDataSeries(), 15)
+        self.instrument = instrument
+        self.sma = ma.SMA(feed[instrument].getAdjCloseDataSeries(), 15)
```
Comment (cicorias) [inline src/features/build_features.py:19]: can we make `15` a parameter on constructor - and the other magic numbers s
```suggestion
        self.sma = ma.SMA(feed[instrument].getAdjCloseDataSeries(), sma_window)
```

File: src/features/build_features.py [hunk 2]
```diff
@@ -1,65 +1,85 @@
...
+    """
+    Construct features (and labels for training) of stock market indicators from daily CSV files
+    using various stock analysis techniques
+
+    Example usage:
+    b = BuildFeatures()
+    X, y = b.run()
+    """
+
     def __init__(self, feed, instrument):
```
Comment (cicorias) [inline src/features/build_features.py:16]: add sma_window = 15
```suggestion
    def __init__(self, feed, instrument, sma_window = 15):
```

File: src/features/build_features.py [hunk 3]
```diff
@@ -1,65 +1,85 @@
...
-        self.__instrument = instrument
-        self.__sma = ma.SMA(feed[instrument].getAdjCloseDataSeries(), 15)
-        self.__ema = ma.EMA(feed[instrument].getAdjCloseDataSeries(), 15)
+        self.instrument = instrument
+        self.sma = ma.SMA(feed[instrument].getAdjCloseDataSeries(), 15)
+        self.ema = ma.EMA(feed[instrument].getAdjCloseDataSeries(), 15)
         self.adj_close = feed[instrument].getAdjCloseDataSeries()
         self.adj_close_slope = linreg.Slope(feed[instrument].getAdjCloseDataSeries(), 15)
-        self.cols = ["Date", "Ticker", "Adj Close", "Adj Close - SMA15", "Adj Close - EMA15"]
+        self.cols = ["Date", "Ticker", "Adj Close", "SMA15", "Adj Close - SMA15", "Adj Close - EMA15"]
```
Comment (cicorias) [inline src/features/build_features.py:23]:
```suggestion
        self.cols = ["Date", "Ticker", "Adj Close", "SMA{}".format(sma_window), "Adj Close - SMA{}".format(sma_window), "Adj Close - EMA{}".format(sma_window)]
```

File: src/features/build_features.py (hunk 1 above)
Comment (cicorias) [reply src/features/build_features.py:19]: ditto for other `15` locations

**Review by cicorias — APPROVED** [review]
LGTM - check some recomendations.

**Comment by cicorias** [comment]
@tedsmoore  -- can you resolve the changes? maybe I merged the other PR too fast.

**Comment by tedsmoore** [comment]
Will do - python notebooks don't play well with Github it seems, every tiny change is a conflict.

**Comment by tedsmoore** [comment]
Ditching PyAlgo. Will make new PR using TA.

**Pull request closed without merging**